File: app/src/main/java/com/emiryanvl/tapreader/domain/usecases/DeleteBookUseCase.java
package com.emiryanvl.tapreader.domain.usecases;

import com.emiryanvl.tapreader.domain.repositories.BookRepository;

import javax.inject.Inject;

public class DeleteBookUseCase {

    private final BookRepository repository;

    @Inject
    public DeleteBookUseCase(BookRepository repository) {
        this.repository = repository;
    }

    public void execute(int id) {
        repository.deleteBook(id);
    }
}

File: app/src/main/java/com/emiryanvl/tapreader/domain/usecases/DeleteBookUseCase.kt
package com.emiryanvl.tapreader.domain.usecases

import com.emiryanvl.tapreader.domain.repositories.BookRepository
import javax.inject.Inject

class DeleteBookUseCase @Inject constructor(private val repository: BookRepository) {
    suspend operator fun invoke(id: Int) = repository.deleteBook(id)
}